ifaddr_private_test.go

```diff
@@ -0,0 +1,100 @@
+package sockaddr
+
+import (
+	"net"
+	"testing"
+)
+
+//Test_getInterfaceIPByFlags is here to facilitate testing of the private function getInterfaceIPByFlags
+func Test_getInterfaceIPByFlags(t *testing.T) {
```

Comment:
my preference (and I may be wrong here) is to test the interface. IOW, let's go thru the exported/ "public" funcs and test all the test cases below? What do you think?

Reply:
I totally agree with you, I would like to test reliably the outer layer too but we cannot control what's returned by `GetAllInterfaces` (called by `GetInterfaceIP` and `GetInterfaceIPWithoutInterfaceFlags`) and consequently, we can only test around the test runner host's interfaces. Unless I am missing something 🤔

I went this way in order to have a quick and dirty way to stub `IfAddrs`  for the various use cases I tested.
Also, I wanted to have a discussion with you guys about the correct way to go about this 😊